central/alert/service/service_impl.go

```diff
@@ -172,8 +172,29 @@ func (s *serviceImpl) GetAlertsCounts(ctx context.Context, request *v1.GetAlerts
 	if request == nil {
 		request = &v1.GetAlertsCountsRequest{}
 	}
+
+	// TODO: @ksanchet - check correctness. This endpoint could possibly be invoked with any query.
+	// For example, alerts within a specific cluster
+	// in that case, will the code below work? The UI only sends in cluster and or namespaces but
+	// an API client can send in any query
 	request.Request = ensureAllAlertsAreFetched(request.GetRequest())
-	alerts, err := s.dataStore.ListAlerts(ctx, request.GetRequest())
+	var requestQ *v1.Query
+	if request.GetRequest().GetQuery() == "" {
+		requestQ = search.EmptyQuery()
+	} else {
+		var err error
+		requestQ, err = search.ParseQuery(request.GetRequest().GetQuery())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	highlightedQ := search.NewQueryBuilder().AddStringsHighlighted(search.Cluster, search.WildcardString).
+		AddStringsHighlighted(search.Category, search.WildcardString).
+		AddStringsHighlighted(search.Severity, search.WildcardString).
+		ProtoQuery()
+
+	alerts, err := s.dataStore.SearchListAlerts(ctx, search.ConjunctionQuery(requestQ, highlightedQ))
```

Comment:
use `Search` that returns `[]search.Result` and collect cluster, severity, and category from the `Matches` field.